client/src/redux/actions/reviewActions.js

```diff
@@ -18,21 +18,28 @@ export const setReviews = (reviews) => {
   };
 
 export const addReview = (newReview) => async(dispatch) => {
-try {
-    const newReviewResponse = await axios.post("/api/reviews", newReview);
-    const newReviewFull = await newReviewResponse.data;
-    return dispatch({
-    type: "ADD_REVIEW",
-    payload: newReviewFull
-    })
-} catch (err) {
-    console.log(err);
-}
+    try {
+        const newReviewResponse = await axios.post("/api/reviews", newReview);
+        const newReviewFull = await newReviewResponse.data;
+        return dispatch({
+        type: "ADD_REVIEW",
+        payload: newReviewFull
+        })
+    } catch (err) {
+        console.log(err);
+    }
 }
 
-export const editReview =  (newReview, index) => {
-    return {
-       type: 'EDIT_REVIEW',
-       payload: { newReview: newReview, index: index},
-    }; 
-};
\ No newline at end of file
+export const editReview =  (newReview, index) => async(dispatch) => {
+    try {
+        const review_id = newReview.review_id;
+        const rating = newReview.rating;
+        axios.put("/api/reviews", {review_id, rating});
+        return dispatch({
+            type: "EDIT_REVIEW",
+            payload: {newReview: newReview, index: index}
```

Comment:
Can this be done without the use of `index`?
After telling the backend which review to update, can we just pull all the fresh reviews?

Reply:
I have this feeling that the Edit/Put flow does not need to be as different from Add/Post as it is now, it can basically be the same thing but idempotent

Reply:
I see...